src/main/scala/query/MatchCreateQuery.scala

```diff
@@ -0,0 +1,74 @@
+package com.originate.scalypher
+
+import action.ReturnAction
+import action.ReturnAll
+import action.ReturnDistinct
+import action.ReturnReference
+import path.AnyNode
+import path.AnyRelationship
+import path.Node
+import path.Path
+import path.Relationship
+import where.Reference
+import types._
+
+trait MatchCreateQuery extends Query {
+
+  def returnAction: Option[ReturnAction]
+
+  def returns(reference: Reference, rest: Reference*): MatchCreateQuery =
+    withReturnAction(ReturnReference(reference, rest: _*))
+
+  def returnDistinct(reference: Reference, rest: Reference*): MatchCreateQuery =
+    withReturnAction(ReturnDistinct(reference, rest: _*))
+
+  def returnAll: MatchCreateQuery =
+    withReturnAction(ReturnAll)
+
+  def getReturnColumns: Set[String] =
+    returnAction match {
+      case Some(action) => matchActionToReturnColumns(action)
+      case _ => Set.empty
+    }
+
+  protected def withReturnAction(action: ReturnAction): MatchCreateQuery
+
+  protected def forcedCreateReferenceables: Set[Referenceable] = Set.empty
+
+  protected def modifiedReferenceableMap: ReferenceableMap = {
+    val returnReferenceables = returnAction map (_.referenceables) getOrElse Set.empty
+    val returnMap = referenceableMap filterKeys { referenceable =>
+      (returnReferenceables contains referenceable) || (forcedCreateReferenceables contains referenceable)
+    }
+    createMap ++ returnMap
+  }
+
+  protected case class PathTranform(path: Path, map: ReferenceableMap = Map[Referenceable, String]())
+
+  protected def cleanedCreatePath: Path
+
+  protected def createMap: ReferenceableMap
+
+  protected def cleanPathAndExtractMap(path: Path, matchPaths: Seq[Path]): (Path, ReferenceableMap) = {
```

Comment:
@a-wortman this is the weird logic i was mentioning... if you want to create a relationship, you do something like: `MATCH (node1 {id: 1}), (node2 {id: 2}) CREATE (node1)-[:RelationshipToCreate]-(node2)`. so in order to continue to allow users to place specific references into their query in order to get identifiers right, i create a "cleaned" version of the create path (otherwise you'd have `CREATE (node1 {id: 1})` since that's how nodes serialize). 

anyways, it got more complicated than i wanted, but it works. let me know if you have any thoughts.